tools/schemapi/schemapi.py

```diff
@@ -218,20 +218,15 @@ def _prepare_refs(d: Dict[str, Any]) -> Dict[str, Any]:
         for key, value in d.items():
             if key == "$ref":
                 d[key] = _VEGA_LITE_ROOT_URI + d[key]
-            else:
-                # $ref values can only be nested in dictionaries or lists
```

Comment:
This comment block seems to have been dropped

Reply:
> All comments should be preserved. Can you please readd them in all of these cases? Maybe just above the `elif` statement that got added.

This case I handled differently to the others in e1ab7667cf29943fd0e38bae14e214b546bf91ba, so I just wanted to add the context before reverting. 


Before [_prepare_refs](https://github.com/vega/altair/blob/18a2c3c237014591d172284560546a2f0ac1a883/tools/schemapi/schemapi.py#L215)
<details>
<summary>Code Block</summary>

```py
def _prepare_references_in_schema(schema: Dict[str, Any]) -> Dict[str, Any]:
    # Create a copy so that $ref is not modified in the original schema in case
    # that it would still reference a dictionary which might be attached to
    # an Altair class _schema attribute
    schema = copy.deepcopy(schema)

    def _prepare_refs(d: Dict[str, Any]) -> Dict[str, Any]:
        """Add _VEGA_LITE_ROOT_URI in front of all $ref values. This function
        recursively iterates through the whole dictionary."""
        for key, value in d.items():
            if key == "$ref":
                d[key] = _VEGA_LITE_ROOT_URI + d[key]
            else:
                # $ref values can only be nested in dictionaries or lists
                # as the passed in `d` dictionary comes from the Vega-Lite json schema
                # and in json we only have arrays (-> lists in Python) and objects
                # (-> dictionaries in Python) which we need to iterate through.
                if isinstance(value, dict):
                    d[key] = _prepare_refs(value)
                elif isinstance(value, list):
                    prepared_values = []
                    for v in value:
                        if isinstance(v, dict):
                            v = _prepare_refs(v)
                        prepared_values.append(v)
                    d[key] = prepared_values
        return d

    schema = _prepare_refs(schema)
    return schema
```

</details>

After [_prepare_refs](https://github.com/dangotbanned/altair/blob/e935ad9a6553828d7bd58bc0b5a6021afa579858/tools/schemapi/schemapi.py#L215)

<details>
<summary>Code Block</summary>

```py
def _prepare_references_in_schema(schema: Dict[str, Any]) -> Dict[str, Any]:
    # Create a copy so that $ref is not modified in the original schema in case
    # that it would still reference a dictionary which might be attached to
    # an Altair class _schema attribute
    schema = copy.deepcopy(schema)

    def _prepare_refs(d: Dict[str, Any]) -> Dict[str, Any]:
        """Add _VEGA_LITE_ROOT_URI in front of all $ref values.

        This function recursively iterates through the whole dictionary.

        $ref values can only be nested in dictionaries or lists
        as the passed in `d` dictionary comes from the Vega-Lite json schema
        and in json we only have arrays (-> lists in Python) and objects
        (-> dictionaries in Python) which we need to iterate through.
        """
        for key, value in d.items():
            if key == "$ref":
                d[key] = _VEGA_LITE_ROOT_URI + d[key]
            elif isinstance(value, dict):
                d[key] = _prepare_refs(value)
            elif isinstance(value, list):
                prepared_values = []
                for v in value:
                    if isinstance(v, dict):
                        v = _prepare_refs(v)
                    prepared_values.append(v)
                d[key] = prepared_values
        return d

    schema = _prepare_refs(schema)
    return schema
```

</details>

I was weighing up the following:
1. The original location of the comment is no longer a distinct `else` branch
	- Whereas the comment describes both `list` and `dict` together
2. This was the only comment within a nested function
	- Unlike a lot of the other comments, this opened the possibility of moving it into the docstring
	- There doesn't seem to be a clear pattern of using comments instead of docstrings for "private"/ `_functions` in `altair`
		- However, if there were, I would assume it wouldn't extend to inner functions that are not exposed as objects to the user 
3. The code is copied from `tools.schemaapi.schemaapi.py` to [altair.utils.schemaapi.py](https://github.com/dangotbanned/altair/blob/e935ad9a6553828d7bd58bc0b5a6021afa579858/altair/utils/schemapi.py#L217)
	- I think at the time I wasn't sure if this was a literal copy of file contents, or something like [ast](https://docs.python.org/3/library/ast.html) - where the latter would have an understanding of the link between the docstring and function
	- After reading through more of `altair` during this PR, I don't believe there's anything relating to abstract syntax trees - so this may be a moot point